src/test/java/com/linecorp/armeria/client/routing/StaticServerGroupTest.java

```diff
@@ -0,0 +1,253 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.routing;
+
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.thrift.ThriftService;
+import com.linecorp.armeria.service.test.thrift.main.HelloService;
+import io.netty.util.internal.PlatformDependent;
+import org.junit.Test;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+public class StaticServerGroupTest {
+    private static final HelloService.Iface HELLO_SERVICE_HANDLER_ONE = (dump) -> "host:127.0.0.1:1234";
+    private static final HelloService.Iface HELLO_SERVICE_HANDLER_TWO = (dump) -> "host:127.0.0.1:2345";
+    private static final HelloService.Iface HELLO_SERVICE_HANDLER_THREE = (dump) -> "host:127.0.0.1:3456";
+
+    private class ServiceServer {
+        private Server server;
+        private HelloService.Iface handler;
+        private int port;
+
+        public ServiceServer(HelloService.Iface handler, int port) {
+            this.handler = handler;
+            this.port = port;
+        }
+
+        private void configureServer() throws Exception {
+            final ServerBuilder sb = new ServerBuilder();
+
+            ThriftService ipService = ThriftService.of(handler);
+
+            sb.serviceAt("/serverIp", ipService);
+            sb.port(this.port, SessionProtocol.HTTP);
+
+            server = sb.build();
+
+            try {
+                server.start().sync();
+            } catch (InterruptedException e) {
+                PlatformDependent.throwException(e);
+            }
+
+        }
+
+        public void start() throws Exception {
+            configureServer();
+        }
+
+        public void stop() {
+            server.stop();
+        }
+    }
+
+    private String nodeAddress(Node node) {
+        return node.hostname() + ":" + node.port();
+    }
+
+
+    @Test
+    public void testRoundRobinSelect() {
+        ServerGroup<WeightedNode> serverGroup = new StaticServerGroup<>(
+                new DefaultWeightedNode("127.0.0.1", 1234),
+                new DefaultWeightedNode("127.0.0.1", 2345),
+                new DefaultWeightedNode("127.0.0.1", 3456));
+
+        ServerGroupRegistry.register("testRoundRobinSelect", serverGroup, NodeSelectStrategy.WEIGHTED_ROUND_ROBIN);
+
+        assertThat(nodeAddress(ServerGroupRegistry.selectNode("testRoundRobinSelect")).equals("127.0.0.1:1234"), is(true));
```

Comment:
Could you use the `TestName` rule? http://stackoverflow.com/questions/473401/get-name-of-currently-executing-test-in-junit-4